test/shopify_app/session/jwt_test.rb

```diff
@@ -50,7 +50,7 @@ class JWTTest < ActiveSupport::TestCase
 
     test "#shopify_domain and #shopify_user_id are nil if the jwt is unsigned" do
       # The library expects there to be a 3rd segment if we want to verify signature
-      expect_jwt_error(::JWT::DecodeError, 'Not enough or too many segments')
+      expect_jwt_error(::JWT::IncorrectAlgorithm, 'Expected a different algorithm')
```

Comment:
I wish the upgrade of `jwt` and this test change could be isolated in their own commit. If I ran across this change later and found that it was part of a commit upgrading omniauth I would be stumped. I realize it might be hard to tease apart the jwt upgrade from the rest of the upgrades, but I just wanted to put my hopes and dreams out there in the world.

Reply:
Yeah I'm trying to see if we can avoid it but because we're allowing patch releases of `jwt` and the bump from `2.2.2` to `2.2.3` included this change, we may have to support it.

Reply:
The gemspec still allows `2.2.2`, which I don't think will work properly after this change? https://github.com/Shopify/shopify_app/pull/1258/files#diff-d72fff1e948b092ced84db03fa37f7c2aa1774033d3aa250d41d847efb2a2264R21

You'll need to require `>= 2.2.3` for this change to be safe, I think.